### PR TITLE
Readable code, function docstring and doctests

### PR DESCRIPTION
**Pull request by Hardvan**
Improved code readability by using separate if-elif statements for conditional checks.
Added inline comments to describe the procedure steps.
Enhanced function documentation for clarity.
Incorporated additional function doctests for thorough testing.
Added Wikipedia reference link for the Fractional Knapsack algorithm.

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Add or change doctests? -- Note: Please avoid changing both code and tests in a single pull request.
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms include at least one URL that points to Wikipedia or another similar explanation.
* [ ] If this pull request resolves one or more open issues then the description above includes the issue number(s) with a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue): "Fixes #ISSUE-NUMBER".

### CODE REVIEW

**Review by cclauss — CHANGES_REQUESTED** [review]
This is checked on the commit message but these changes do NOT comply.
## Please avoid changing both code and tests in a single pull request.

**Comment by Hardvan** [comment]
I have removed the added doctests. Now, there is only one change - the code.

**Comment by cclauss** [comment]
Great.  Please open another pull request with the doctests and we can merge that one before this one.

**Comment by Hardvan** [comment]
I have opened the PR #10891 which has the added doctests. Thanks!

**Comment by cclauss** [comment]
OK.  Please add the tests that have been merged to `master`.
* https://github.com/TheAlgorithms/Python/pull/10891/files

**Comment by Hardvan** [comment]
Added the tests that have been merged to master.

File: greedy_methods/fractional_knapsack.py [hunk 1]
```diff
@@ -32,17 +45,18 @@ def frac_knapsack(vl, wt, w, n):
     TypeError: unsupported operand type(s) for /: 'str' and 'int'
     """
 
+    # sort in descending order of value/weight ratio
     r = sorted(zip(vl, wt), key=lambda x: x[0] / x[1], reverse=True)
```
Comment (cclauss) [inline greedy_methods/fractional_knapsack.py:49]: Please add this and fix the tests to match.
```suggestion
    r = sorted(zip(vl, wt), key=lambda x: x[0] / x[1], reverse=True, strict=True)
```

File: greedy_methods/fractional_knapsack.py [hunk 2]
```diff
@@ -1,9 +1,22 @@
 from bisect import bisect
 from itertools import accumulate
 
+# Reference Link: https://en.wikipedia.org/wiki/Continuous_knapsack_problem
 
-def frac_knapsack(vl, wt, w, n):
+
+def frac_knapsack(vl: list, wt: list, w: int, n: int) -> float:
```
Comment (cclauss) [inline greedy_methods/fractional_knapsack.py:7]: Let's use self-documenting variable names...
```suggestion
def frac_knapsack(values: list, weights: list, capacity: int, max_items: int) -> float:
```

**Comment by Hardvan** [comment]
Renamed the variables.
Adding strict=True produces this error:
TypeError: sort() takes at most 2 keyword arguments (3 given)

File: greedy_methods/fractional_knapsack.py [hunk 3]
```diff
@@ -1,19 +1,38 @@
...
+        ...
+    ValueError: zip() argument 2 is shorter than argument 1
     >>> frac_knapsack([10, 40, 30], [5, 4, 6, 3], 8, 4)
-    60.0
+    Traceback (most recent call last):
+        ...
+    ValueError: zip() argument 2 is longer than argument 1
     >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6, 3], 0, 4)
     0
     >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6, 3], 8, 0)
```
Comment (cclauss) [inline greedy_methods/fractional_knapsack.py:38]: Should we raise a `ValueError` if `capacity` is <= 0?  That would seem to me to be garbage data.

Our code does not consider `max_items` at all!

**Comment by Hardvan** [comment]
I'll add the changes tomorrow. Thanks for highlighting the issues.

**Comment by Hardvan** [comment]
I have added the input validation check for `capacity` and `max_items` variables, and updated the doctests accordingly.

File: greedy_methods/fractional_knapsack.py [hunk 4]
```diff
@@ -1,27 +1,54 @@
...
+    Returns:
+        Maximum value of items that can be put into the knapsack.
+
     >>> frac_knapsack([60, 100, 120], [10, 20, 30], 50, 3)
     240.0
     >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6, 3], 10, 4)
     105.0
     >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6, 3], 8, 4)
     95.0
     >>> frac_knapsack([10, 40, 30, 50], [5, 4, 6], 8, 4)
```
Comment (cclauss) [inline greedy_methods/fractional_knapsack.py:28]: This looking much better but is the _fractional_ part working?

https://www.geeksforgeeks.org/fractional-knapsack-problem says that
```
>>> frac_knapsack([500], [3], 10, 1)
166.66667
```
but we get `500`

Comment (Hardvan) [reply greedy_methods/fractional_knapsack.py:28]: You are right... the existing algo does not implement fractional knapsack correctly.

**Comment by Hardvan** [comment]
I have corrected the algorithm.

**Pull request closed without merging**